libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxDistanceFormatter.kt

```diff
@@ -81,51 +76,40 @@ class MapboxDistanceFormatter(
      * relative size of .65 times the size of the number
      */
     override fun formatDistance(distance: Double): SpannableString {
-        val distanceSmallUnit =
-            TurfConversion.convertLength(distance, UNIT_METERS, smallUnit)
-        val distanceLargeUnit =
-            TurfConversion.convertLength(distance, UNIT_METERS, largeUnit)
-
-        return when {
-            // If the distance is greater than 10 miles/kilometers, then round to nearest mile/kilometer
-            distanceLargeUnit > LARGE_UNIT_THRESHOLD -> {
-                getDistanceString(roundToDecimalPlace(distanceLargeUnit, 0), largeUnit)
+        val distanceAndSuffix = when (distance) {
+            in 0.0..smallDistanceUpperThreshold -> {
+                getDistanceAndSuffixForSmallUnit(distance)
             }
-            // If the distance is less than 401 feet/meters, round by fifty feet/meters
-            distanceSmallUnit < SMALL_UNIT_THRESHOLD -> {
-                getDistanceString(roundToClosestIncrement(distanceSmallUnit), smallUnit)
+            in mediumDistanceLowerThreshold..mediumDistanceUpperThreshold -> {
+                getDistanceAndSuffixForLargeUnit(distance, 1)
             }
-            // If the distance is between 401 feet/meters and 10 miles/kilometers, then round to one decimal place
             else -> {
-                getDistanceString(roundToDecimalPlace(distanceLargeUnit, 1), largeUnit)
+                getDistanceAndSuffixForLargeUnit(distance, 0)
             }
         }
+        return getSpannableDistanceString(distanceAndSuffix)
     }
 
-    /**
-     * Returns number rounded to closest specified rounding increment, unless the number is less than
-     * the rounding increment, then the rounding increment is returned
-     *
-     * @param distance to round to closest specified rounding increment
-     * @return number rounded to closest rounding increment, or rounding increment if distance is less
-     */
-    private fun roundToClosestIncrement(distance: Double): String {
-        val roundedNumber = Math.round(distance).toInt() / roundingIncrement * roundingIncrement
-
-        return (if (roundedNumber < roundingIncrement) roundingIncrement else roundedNumber).toString()
+    private fun getDistanceAndSuffixForSmallUnit(distance: Double): Pair<String, String> {
```

Comment:
IMO get / set nomenclature is hijacked by the Java language and should be only used when defining POJOs / DTOs.  The same thing applies to some other functions below 👇